### PR TITLE
types/views,cmd/viewer: add ByteSlice[T] to replace mem.RO

### PR DESCRIPTION
**Pull request by maisem**
Add a new views.ByteSlice[T ~[]byte] to provide a better API to use with views.

Updates #cleanup

### CODE REVIEW

File: types/views/views.go [hunk 1]
```diff
@@ -21,6 +24,80 @@ func unmarshalSliceFromJSON[T any](b []byte, x *[]T) error {
...
+func (v ByteSlice[T]) Equal(b T) bool {
+	return bytes.Equal(v.ж, b)
+}
+
+// EqualView reports whether the underlying slice is equal to b.
+func (v ByteSlice[T]) EqualView(b ByteSlice[T]) bool {
+	return bytes.Equal(v.ж, b.ж)
+}
+
+// AsSlice returns a deep-copy of the underlying slice.
```
Comment (bradfitz) [inline types/views/views.go:66]: "deep"? It's a byte slice by definition.

Comment (maisem) [reply types/views/views.go:66]: Copy paste fail, will fix

File: types/views/views.go [hunk 2]
```diff
@@ -21,6 +24,80 @@ func unmarshalSliceFromJSON[T any](b []byte, x *[]T) error {
...
+func (v ByteSlice[T]) AsSlice() T {
+	return v.AppendTo(v.ж[:0:0])
+}
+
+// AppendTo appends the underlying slice values to dst.
```
Comment (bradfitz) [inline types/views/views.go:71]: You have all of Mem.RO almost but not At. Add At too?

**Review by bradfitz — COMMENTED** [review]
Nice.

**Review by bradfitz — APPROVED** [review]

**Comment by twitchyliquid64** [comment]
Yeah I like this better, cheers

**Pull request merged**